libraries/AP_ICEngine/AP_ICEngine.h

```diff
@@ -25,6 +25,13 @@ class AP_ICEngine {
     // constructor
     AP_ICEngine(const AP_RPM &_rpm);
 
+    /* Do not allow copies */
+    AP_ICEngine(const AP_ICEngine &other) = delete;
+    AP_ICEngine &operator=(const AP_ICEngine&) = delete;
+
+    // get singleton
+    static AP_ICEngine *get_instance();
```

Comment:
This should definitely be renamed to `get_singeton`.  The naming within the class is less critical, but we have certainly converged on `get_singleton` for the getter.